gdpr/impl.go

```diff
@@ -63,53 +60,32 @@ func (p *permissionsImpl) AuctionActivitiesAllowed(ctx context.Context, bidderCo
 	if _, ok := p.nonStandardPublishers[p.publisherID]; ok {
 		return AllowAll, nil
 	}
-
 	if p.gdprSignal != SignalYes {
 		return AllowAll, nil
 	}
-
 	if p.consent == "" {
 		return p.defaultPermissions(), nil
 	}
-
-	// note the bidder here is guaranteed to be enabled
-	vendorID, vendorFound := p.resolveVendorID(bidderCoreName, bidder)
-	basicEnforcementVendors := p.cfg.BasicEnforcementVendors()
-	_, weakVendorEnforcement := basicEnforcementVendors[string(bidder)]
-
-	if !vendorFound && !weakVendorEnforcement {
-		return DenyAll, nil
-	}
-
-	parsedConsent, vendor, err := p.parseVendor(ctx, vendorID, p.consent)
+	pc, err := parseConsent(p.consent)
 	if err != nil {
 		return p.defaultPermissions(), err
 	}
-
-	// vendor will be nil if not a valid TCF2 consent string
-	if vendor == nil {
-		if weakVendorEnforcement && parsedConsent.Version() == 2 {
-			vendor = vendorTrue{}
-		} else {
-			return p.defaultPermissions(), nil
-		}
-	}
-
```

Comment:
If we were unable to find the GVL ID for the bidder when we called `resolveVendorID` above and we made it this far without returning early, that meant that the bidder was listed as a soft vendor exception. This call to `parseVendor` would end up returning nil for `vendor` which would either cause an early return of the default permissions on line 94 if the version was incorrect in the consent string or would set `vendor` to `vendorTrue`. Setting it to `vendorTrue` is unnecessary because the basic enforcement algorithm would be selected by the purpose enforcer builder since the bidder is listed as a soft vendor exception (basic enforcement vendor in the config) and the basic enforcement algorithm looks for and properly handles the soft vendor exception.